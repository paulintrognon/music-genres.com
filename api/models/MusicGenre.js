const Sequelize = require('sequelize');
const sequelize = require('../db.js').sequelize;

const MusicGenre = sequelize.define('music_genre', {
  name: {
    type: Sequelize.STRING,
  },
});

MusicGenre.sync();

module.exports = MusicGenre;
