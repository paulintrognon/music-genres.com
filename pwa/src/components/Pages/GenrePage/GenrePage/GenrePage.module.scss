.tracks {
  margin-top: 3rem;
}

.addContainer {
  margin-top: 1.5rem;
  text-align: center;
}
