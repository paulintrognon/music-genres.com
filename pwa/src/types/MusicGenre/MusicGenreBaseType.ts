export type MusicGenreBaseType = {
  id: number
  name: string
  slug: string
}
