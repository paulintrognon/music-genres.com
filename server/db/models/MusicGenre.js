const Sequelize = require('sequelize');
const sequelize = require('../db.js').sequelize;

const MusicGenre = sequelize.define('musicGenre', {
  name: {
    type: Sequelize.STRING,
    allowNull: false,
    notEmpty: true,
    len: [2, 100],
    unique: true,
  },
  slug: {
    type: Sequelize.STRING,
    allowNull: false,
    notEmpty: true,
    len: [2, 100],
    unique: true,
  },
});

module.exports = MusicGenre;
