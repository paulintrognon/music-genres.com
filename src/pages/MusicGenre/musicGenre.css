.music-genre-page .top-tracks-container {
  margin-top: 80px;
}

.music-genre-page .add-track-row {
  margin-top: 108px;
  text-align: center;
}

.music-genre-page .more-of-title {
  margin-top: 100px;
}

.music-genre-page .other-tracks-container {
  margin-top: 100px;
}

@media (max-width: 700px) {
  .music-genre-page .top-tracks-container,
  .music-genre-page .other-tracks-container {
    margin-top: 30px;
    margin-left: -10px;
    margin-right: -10px;
  }

  .music-genre-page .add-track-row,
  .music-genre-page .more-of-title {
    margin-top: 50px;
  }
}
