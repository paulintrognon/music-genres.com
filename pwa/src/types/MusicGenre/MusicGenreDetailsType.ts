import { TrackType } from '../Track/TrackType'
import { MusicGenreBaseType } from './MusicGenreBaseType'

export type MusicGenreDetailsType = MusicGenreBaseType & {
  tracks: TrackType[]
}
