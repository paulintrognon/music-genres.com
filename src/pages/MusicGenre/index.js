import React from 'react';
import { connect } from 'react-redux';

import {
  goToAddVideoToGenre
} from '../../actions/navigationActions';
import {
  fetchMusicGenreAction,
  voteForTrackAction,
} from '../../actions/musicGenreActions';
import {
  playTrackAction,
} from '../../actions/playerActions';

import HashtagTitle from '../../components/HashtagTitle';
import LoadingIndicator from '../../components/LoadingIndicator';
import Plus from '../../components/images/Plus';
import RectangleButton from '../../components/RectangleButton';
import Track from './Track';

import './musicGenre.css';

function mapStoreToProps(store) {
  return store.musicGenre;
}
class MusicGenre extends React.Component {
  componentWillMount() {
    this.props.dispatch(fetchMusicGenreAction(this.props.match.params.slug));
  }

  voteForTrack = (track) => {
    this.props.dispatch(voteForTrackAction(track, this.props.musicGenre.id));
  }

  playTrack = (trackIndex) => {
    this.props.dispatch(playTrackAction(trackIndex, this.props.musicGenre));
  }

  addVideoToThisGenre = () => {
    this.props.dispatch(goToAddVideoToGenre(this.props.musicGenre.slug, true));
  }

  render() {
    return (
      <div className="music-genre-page">
        {this.renderContent()}
      </div>
    );
  }

  renderContent = () => {
    if (this.props.isFetching) {
      return <LoadingIndicator></LoadingIndicator>;
    }
    if (this.props.error) {
      return <p>Erreur !</p>;
    }
    if (this.props.isFetched) {
      return <div>
        <HashtagTitle>{this.props.musicGenre.name}</HashtagTitle>
        {this.renderTracks(this.props.musicGenre.tracks)}
      </div>;
    }
    return null;
  }

  renderTracks = (tracks) => {
    if (!tracks.length) {
      return <p>No tracks yet.</p>
    }
    const otherTracks = tracks.slice(3);
    return (
      <div>
        <div className="top-tracks-container">
          {[0,1,2].map(index => (
            <Track key={index}
              track={tracks[index]}
              index={index}
              onVoteHandler={() => this.voteForTrack(tracks[index])}
              onPlayHandler={() => this.playTrack(index)}
              ></Track>
          ))}
        </div>
        {this.renderAddTrack()}
        <HashtagTitle className="more-of-title">More from {this.props.musicGenre.name}</HashtagTitle>
        <div className="other-tracks-container">
          {otherTracks.map((track, i) => (
            <Track key={i}
              track={track}
              onVoteHandler={() => this.voteForTrack(track)}
              onPlayHandler={() => this.playTrack(i+3)}
              ></Track>
          ))}
        </div>
        {this.renderAddTrack()}
      </div>
    );
  }

  renderAddTrack = () => {
    return (
      <div className="add-track-row">
        <RectangleButton onClick={this.addVideoToThisGenre}>
          Add video&nbsp;<Plus></Plus>
        </RectangleButton>
      </div>
    );
  }
}
export default connect(mapStoreToProps)(MusicGenre);
