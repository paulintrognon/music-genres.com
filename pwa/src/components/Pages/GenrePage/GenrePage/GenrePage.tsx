import Link from 'next/link'
import { MusicGenreDetailsType } from '../../../../types/MusicGenre/MusicGenreDetailsType'
import SquareButton from '../../../Atoms/Buttons/SquareButton/SquareButton'
import HashtagTitle from '../../../Atoms/Titles/HashtagTitle/HashtagTitle'
import PlusIcon from '../../../Atoms/Icons/PlusIcon/PlusIcon'
import TrackCard from '../TrackCard/TrackCard'
import classes from './GenrePage.module.scss'

interface Props {
  genre: MusicGenreDetailsType
}
const GenrePage: React.FC<Props> = ({ genre }) => {
  return (
    <div className={classes.container}>
      {/* Heading */}
      <HashtagTitle>{genre.name}</HashtagTitle>

      {/* Tracks listing */}
      <div className={classes.tracks}>
        <div className="row">
          {genre.tracks.map((track) => (
            <div key={track.id} className="col-xs-4">
              <TrackCard track={track} />
            </div>
          ))}
        </div>
      </div>

      {/* Add track */}
      <div className={classes.addContainer}>
        <Link href={`/add/video/${genre.slug}`}>
          <a>
            <SquareButton>
              <PlusIcon />
              Add video
            </SquareButton>
          </a>
        </Link>
      </div>
    </div>
  )
}
export default GenrePage
