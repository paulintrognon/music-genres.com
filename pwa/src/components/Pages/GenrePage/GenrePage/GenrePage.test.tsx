import { shallow } from 'enzyme'
import { MusicGenreDetailsType } from '../../../../types/MusicGenre/MusicGenreDetailsType'
import GenrePage from './GenrePage'

it('should render the GenrePage', () => {
  const genre: MusicGenreDetailsType = {
    id: 1,
    name: 'Super Genre',
    slug: 'super-genre',
    tracks: [{ id: 1, title: 'Super Title', playerTrackId: 'abc123', playerName: 'youtube' }],
  }
  const wrapper = shallow(<GenrePage genre={genre} />)
  expect(wrapper).toMatchSnapshot()
})
