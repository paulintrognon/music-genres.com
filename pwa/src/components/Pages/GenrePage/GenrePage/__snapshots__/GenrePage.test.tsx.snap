// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the GenrePage 1`] = `
<div
  className="container"
>
  <HashtagTitle>
    Super Genre
  </HashtagTitle>
  <div
    className="tracks"
  >
    <div
      className="row"
    >
      <div
        className="col-xs-4"
        key="1"
      >
        <TrackCard
          track={
            Object {
              "id": 1,
              "playerName": "youtube",
              "playerTrackId": "abc123",
              "title": "Super Title",
            }
          }
        />
      </div>
    </div>
  </div>
  <div
    className="addContainer"
  >
    <Link
      href="/add/video/super-genre"
    >
      <a>
        <SquareButton>
          <PlusIcon />
          Add video
        </SquareButton>
      </a>
    </Link>
  </div>
</div>
`;
